### PR TITLE
Added tests around the voting ballot contract.

### LINKED ISSUE #488: [Solidity Support] Voting example
### Problem

The Hedera network supports solidity as one of the smart contracts development languages.
Occasionally items have shown up that surfaced in-equivalence at the language layer.

### Solution

Implement [Voting example](https://docs.soliditylang.org/en/latest/solidity-by-example.html#voting) with supporting tests 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ebadiere**
Implementation of the Voting example

**Related issue(s)**:

Fixes #488 

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/solidity/voting/Ballot.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+
+describe('@solidityequiv1 Ballot Units tests', function () {
```
Comment (Nana-EC) [inline test/solidity/voting/Ballot.js:24]: Let's put all the examples in one group
```suggestion
describe('@solidityequiv4 Ballot Units tests', function () {
```

File: test/solidity/voting/Ballot.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    
+    it("Should allow voting for a proposal", async function() {
+        await ballotContract.giveRightToVote(addressB.address);
+        await ballotContract.connect(addressB).vote(1); // voting for proposal2
+        const proposal = await ballotContract.proposals(1);
+        expect(proposal.voteCount).to.equal(1);
+    });
+
+    it("Should correctly determine the winning proposal", async function() {
+        await ballotContract.giveRightToVote(addressB.address);
```
Comment (Nana-EC) [inline test/solidity/voting/Ballot.js:61]: Q: shouldn't there be multiple proposals to make sure that the logic works correctly?

File: hardhat.config.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const {
   HEX_PRIVATE_KEY_A,
   HEX_PRIVATE_KEY_B,
   NETWORKS,
+  HEX_PRIVATE_KEY_C,
```
Comment (Nana-EC) [inline hardhat.config.js:11]: Please update `TEST_SETUP.md`

**Review by Nana-EC — CHANGES_REQUESTED** [review]

File: test/solidity/voting/Ballot.js (hunk 1 above)
Comment (ebadiere) [reply test/solidity/voting/Ballot.js:24]: Done.

File: test/solidity/voting/Ballot.js (hunk 2 above)
Comment (ebadiere) [reply test/solidity/voting/Ballot.js:62]: Added more proposals and votes.

File: hardhat.config.js (hunk 3 above)
Comment (ebadiere) [reply hardhat.config.js:11]: Done.

File: TEST_SETUP.md [hunk 4]
```diff
@@ -30,8 +30,12 @@
    | `OPERATOR_KEY_A`    | This is the `DER Encoded Private Key` which can be found in your account portal. |
    | `HEX_PRIVATE_KEY_A` | This is the `HEX Encoded Private Key` which can be found in your account portal. |
    | `HEX_PRIVATE_KEY_B` | This is the `HEX Encoded Private Key` which can be found in your account portal. |
+   | `HEX_PRIVATE_KEY_C` | This is the `HEX Encoded Private Key` which can be found in your account portal. This key is used in the voting example. |
```
Comment (Nana-EC) [inline TEST_SETUP.md:33]: This isn't sustainable.
At this point this env variable should be an array.
Can you explore using a `HEX_PRIVATE_KEYS` that holds a list of accounts

File: contracts/solidity/voting/Ballot.sol [hunk 5]
```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0
```
Comment (Nana-EC) [inline contracts/solidity/voting/Ballot.sol:1]: All licenses should be Apache 2.0

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good but let's improve the accounts env variable layout

File: contracts/solidity/voting/Ballot.sol (hunk 5 above)
Comment (ebadiere) [reply contracts/solidity/voting/Ballot.sol:1]: Done.

File: TEST_SETUP.md (hunk 4 above)
Comment (ebadiere) [reply TEST_SETUP.md]: Done.

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**